### DISCUSSION TITLE
Withdrawal Script Research

### DISCUSSION THREAD

**Discussion by setzeus** [stacks-sbtc/sbtc#7] [Research]
The current design for withdrawing from sBTC -> BTC involves four Clarity contracts with the ability for users to submit *either* a BTC or STX transaction to initiate a withdrawal. 

The five contracts involved are:

1. sbtc-registry -> tracks all pending withdrawal requests
2. sbtc-withdrawal-request-btc -> initiates withdrawal from burnchain
3. sbtc-withdrawal-request-stx -> initiates withdrawal from stacks
4. sbtc-withdrawal-request-reclaim -> called to claw back / withdraw a withdrawal
5. sbtc-withdrawal-verifier -> called to verify a withdrawal & burn the locked-sbtc

The item in question, the script involved in sbtc-withdrawal-request-btc is the following:

**magic (2 bytes) + op (1 byte) + amount (8 bytes) + schnorr signature (64 bytes)**

The magic bytes are what clients will use to detect that this is an sBTC operation, the op byte is op_return, the amount is numerical value of satoshis withdrawn which can be up to eight bytes & the final item is the schnorr signature for the corresponding p2tr output.

**Comment by netrome** [comment]
I'm under the impression that we won't support the BTC-initiated withdrawal request. At least initially. I'd love to hear if there is any arguments for supporting it, otherwise the flow feels much simpler if it's just initiated as a Stacks contract call.
